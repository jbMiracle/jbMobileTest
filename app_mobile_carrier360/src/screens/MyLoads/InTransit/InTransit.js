import React, { Component } from 'react';
import { View } from 'react-native';
import { Text, Button } from 'native-base';
import C360Colors from '../../../themes/360Colors';

class InTransit extends Component {
  render() {
    const { params } = this.props.navigation.state;
    const msg = params !== undefined ? params.msg : 'Water those.';
    return (
      <View testID='in_transit' style={ { flex: 1, justifyContent: 'center', alignItems: 'center' } }>
        <Text>In Transit</Text>
        <Button
          style={ {
              backgroundColor: C360Colors.Yellow500, padding: 0, marginTop: 15, alignSelf: 'center',
          } }
          onPress={ () => this.props.navigation.navigate('Load Details', { msg }) }
        >
          <Text style={ { color: C360Colors.Blue600, fontSize: 12, fontWeight: 'bold' } }>Show Load Details</Text>
        </Button>
      </View>
    );
  }
}

export default InTransit;
