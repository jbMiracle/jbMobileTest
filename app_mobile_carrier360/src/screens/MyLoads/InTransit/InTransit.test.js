import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import InTransit from '../InTransit/InTransit';

it('can render an empty component', () => {
  const renderer = new ShallowRenderer();
  const props = {
    navigation: {
      state: {
        params: {
          msg: 'test',
        },
      },
    },
  };
  renderer.render(<InTransit { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
