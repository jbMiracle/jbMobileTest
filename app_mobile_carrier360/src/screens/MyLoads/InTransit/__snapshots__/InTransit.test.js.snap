// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render an empty component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="in_transit"
>
  <Styled(Text)
    virtual={undefined}
  >
    In Transit
  </Styled(Text)>
  <Styled(Button)
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFDB00",
        "marginTop": 15,
        "padding": 0,
      }
    }
    virtual={undefined}
  >
    <Styled(Text)
      style={
        Object {
          "color": "#005F8E",
          "fontSize": 12,
          "fontWeight": "bold",
        }
      }
      virtual={undefined}
    >
      Show Load Details
    </Styled(Text)>
  </Styled(Button)>
</View>
`;
